src/lib/pages/nexus-endpoints.svelte

```diff
@@ -1,121 +1,179 @@
     routeForNexusEndpoint,
-    routeForNexusEndpointCreate,
   } from '$lib/utilities/route-for';
   import { updateQueryParameters } from '$lib/utilities/update-query-parameters';
 
-  export let endpoints: NexusEndpoint[] = [];
-  export let searchPlaceholder = translate('common.search');
-  export let createDisabled = false;
+  type Props = {
+    endpoints: NexusEndpoint[];
+    searchPlaceholder?: string;
+    createDisabled?: boolean;
+    createHref?: string;
+    headerColumns?: Snippet;
+    rowColumns?: Snippet<[NexusEndpoint]>;
+  };
+
+  let {
+    endpoints = [],
+    searchPlaceholder = translate('common.search'),
+    createDisabled = false,
+    createHref = '/nexus/create',
+    headerColumns,
+    rowColumns,
+  }: Props = $props();
 
-  let search = '';
-  $: searchParam = $page.url.searchParams.get('search') || '';
+  let search = $state('');
+  let searchParam = $derived(page.url.searchParams.get('search') || '');
 
   const searchParamUpdate = debounce((value) => {
     updateQueryParameters({
       parameter: 'search',
       value,
-      url: $page.url,
+      url: page.url,
     });
   }, 350);
 
-  $: {
+  $effect(() => {
     searchParamUpdate(search);
-  }
+  });
 </script>
 
 {#if !endpoints?.length && !searchParam}
-  <NexusEmptyState {createDisabled}>
-    <slot />
-  </NexusEmptyState>
+  <NexusEmptyState {createDisabled} {createHref} />
 {:else}
   <div class="mb-8 flex items-center justify-between">
     <h1 data-testid="namespace-selector-title">
       {translate('nexus.endpoints')}
     </h1>
-    <Button
-      disabled={createDisabled}
-      variant="primary"
-      href={routeForNexusEndpointCreate()}
+    <Button disabled={createDisabled} variant="primary" href={createHref}
       >{translate('nexus.create-endpoint')}</Button
     >
   </div>
   <div class="flex flex-col gap-4">
-    <div class="flex flex-col justify-between lg:flex-row">
-      <Input
-        id="endpoint-search"
-        bind:value={search}
-        icon="search"
-        label={searchPlaceholder}
-        labelHidden
-        autoFocus
-        type="search"
-        placeholder={searchPlaceholder}
-        class="w-full lg:w-1/2"
-      />
-    </div>
+    <Input
+      id="endpoint-search"
+      bind:value={search}
+      icon="search"
+      label={searchPlaceholder}
+      labelHidden
+      autoFocus
+      type="search"
+      placeholder={searchPlaceholder}
+      class="w-full"
+    />
     {#if endpoints.length}
-      <div
-        class="grid grid-cols-1 gap-4 pr-8 md:grid-cols-2 md:pr-24 lg:grid-cols-3 xl:grid-cols-4 xl:pr-48"
-      >
+      <Table class="w-full" bordered>
+        <caption class="sr-only" slot="caption">
+          {translate('nexus.endpoints')}
+        </caption>
+        <TableHeaderRow slot="headers">
+          <th>Name</th>
+          <th>Used By</th>
+          {@render headerColumns?.()}
+          <th>Last Updated</th>
+          <th>Created On</th>
+        </TableHeaderRow>
```

Comment:
It might be a good idea to pass in the full TableHeaderRow in the headerColumns snippet instead of having it in the middle of hardcoded headers. That way you have more flexibility in the future and can keep OSS/Cloud separation clean.